vital/config/tests/test_config_block.cxx

```diff
@@ -901,3 +903,34 @@ IMPLEMENT_TEST(set_value_description)
       "accessing description of unset key"
       );
 }
+
+
+// Test macro
+ENUM_CONVERTER( my_ec, int,
+          // init stuff
+          { "one",   1 },
+          { "two",   2 },
+          { "three", 3 },
+          { "four",  4 },
+          { "five",  5 }
+  )
+
+
+// ------------------------------------------------------------------
+IMPLEMENT_TEST(enum_conversion)
+{
+  using namespace kwiver::vital;
+
+  config_block_sptr const config = config_block::empty_config();
+
+  config->set_value("keya", "three");
+  config->set_value("keyb", "foo");
+
+  int val = config->get_enum_value < my_ec >( "keya" );
+  TEST_EQUAL("enum value", val, 3 );
+
+  EXPECT_EXCEPTION( std::runtime_error,
+                    config->get_enum_value < my_ec >( "keyb" ),
+                    "enum conversion error");
+
```

Comment:
Maybe add another expect_exception for trying to retrieve a non-existent key (e.g. get_enum_value<my_ec>("foo"))